pkg/graph/drive.resolvers.go

```diff
@@ -424,6 +457,142 @@ func (r *mutationResolver) CreateSharedDrive(ctx context.Context, name string) (
 	return bucket, nil
 }
 
+// CreateShare is the resolver for the createShare field.
+func (r *mutationResolver) CreateShare(ctx context.Context, input model.CreateShareInput) (*model.ShareUser, error) {
+	currentUser, err := middleware.GetUser(ctx)
+	if err != nil {
+		return nil, nil
+	}
+
+	// Get user
+	var user db.User
+	err = r.DB.NewSelect().Model(&user).Where("id = ?", input.User).Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if we have fileId or bucketId
+	if input.FileID == nil && input.BucketID == nil {
+		return nil, errors.New("fileId or bucketId is required")
+	}
+
+	share := db.Share{
+		SharedWith:     input.User,
+		SharedBy:       currentUser.ID,
+		OrganisationID: currentUser.OrganisationID,
+		Permission:     strings.ToLower(string(input.Permission)),
+	}
+
+	if input.FileID != nil {
+		share.FileID = *input.FileID
+	} else {
+		share.BucketID = *input.BucketID
+	}
+
+	err = r.DB.NewInsert().Model(&share).Returning("*").Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &model.ShareUser{
+		User:       &user,
+		Permission: input.Permission,
+	}, nil
+}
+
+// EditShare is the resolver for the editShare field.
+func (r *mutationResolver) EditShare(ctx context.Context, input model.CreateShareInput) (*model.ShareUser, error) {
+	currentUser, err := middleware.GetUser(ctx)
+	if err != nil {
+		return nil, nil
+	}
+
+	// Get share
+	var share db.Share
+	if input.FileID != nil {
+		err = r.DB.NewSelect().Model(&share).
+			Where("shared_with= ?", input.User).
+			Where("organisation_id = ?", currentUser.OrganisationID).
+			Where("file_id = ?", input.FileID).
+			Scan(ctx)
+	} else if input.BucketID != nil {
+		err = r.DB.NewSelect().Model(&share).
+			Where("shared_with= ?", input.User).
+			Where("organisation_id = ?", currentUser.OrganisationID).
+			Where("bucket_id = ?", input.BucketID).
+			Scan(ctx)
+	} else {
+		return nil, errors.New("fileId or bucketId is required")
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	// Update permission
+	share.Permission = strings.ToLower(string(input.Permission))
```

Comment:
Is there a cleaner way to handle the parsing of `input.Permission`? Perhaps type the input?